GVFS/GVFS/CommandLine/StatisticsVerb.cs

```diff
@@ -0,0 +1,209 @@
+        /// <returns>An array containing all of the modified paths in string format</returns>
+        private string[] GetModifiedPathsFromPipe(GVFSEnlistment enlistment)
+        {
+            using (NamedPipeClient pipeClient = new NamedPipeClient(enlistment.NamedPipeName))
+            {
+                string[] modifiedPathsList = { };
+
+                if (!pipeClient.Connect())
+                {
+                    this.ReportErrorAndExit("Unable to connect to GVFS.  Try running 'gvfs mount'");
+                }
+
+                try
+                {
+                    NamedPipeMessages.Message modifiedPathsMessage = new NamedPipeMessages.Message(NamedPipeMessages.ModifiedPaths.ListRequest, "1");
+                    pipeClient.SendRequest(modifiedPathsMessage);
+
+                    NamedPipeMessages.Message modifiedPathsResponse = pipeClient.ReadResponse();
+                    if (!modifiedPathsResponse.Header.Equals(NamedPipeMessages.ModifiedPaths.SuccessResult))
+                    {
+                        this.Output.WriteLine("Bad response from modified path pipe: " + modifiedPathsResponse.Header);
+                        return modifiedPathsList;
+                    }
+
+                    modifiedPathsList = modifiedPathsResponse.Body.Split('\0');
+                    modifiedPathsList = modifiedPathsList.Take(modifiedPathsList.Length - 1).ToArray();
+                }
+                catch (BrokenPipeException e)
+                {
+                    this.ReportErrorAndExit("Unable to communicate with GVFS: " + e.ToString());
+                }
+
+                return modifiedPathsList;
+            }
+        }
+
+        /// <summary>
+        /// Get two lists of placeholders, one containing the files and the other the directories
+        /// Goes to the SQLite database for the placeholder lists
+        /// </summary>
+        /// <param name="enlistment">The current GVFS enlistment being operated on</param>
+        /// <param name="filePlaceholders">Out parameter where the list of file placeholders will end up</param>
+        /// <param name="folderPlaceholders">Out parameter where the list of folder placeholders will end up</param>
+        private void GetPlaceholdersFromDatabase(GVFSEnlistment enlistment, out List<IPlaceholderData> filePlaceholders, out List<IPlaceholderData> folderPlaceholders)
+        {
+            GVFSDatabase database = new GVFSDatabase(new PhysicalFileSystem(), enlistment.EnlistmentRoot, new SqliteDatabase());
+            PlaceholderTable placeholderTable = new PlaceholderTable(database);
+
+            placeholderTable.GetAllEntries(out filePlaceholders, out folderPlaceholders);
+        }
+
+        private List<string> GetPathsFromGitIndex(GVFSEnlistment enlistment)
+        {
+            List<string> gitPaths = new List<string>();
+            GitProcess gitProcess = new GitProcess(enlistment);
+
+            GitProcess.Result result = gitProcess.LsTree(

```

Comment:
I'm wondering if this should instead be using `git ls-files -sv` so that we can get what the skipworktree bit is and the number of files that git is keeping up to date and the number that should be being projected.

Reply:
See my reply to the comment 2 after this one